src/Providers/Http/DTO/Request.php

```diff
@@ -281,6 +297,104 @@ public function getData(): ?array
         return $this->data;
     }
 
+    /**
+     * Gets the request options.
+     *
+     * @since n.e.x.t
+     *
+     * @return RequestOptions|null Request transport options when configured.
+     */
+    public function getOptions(): ?RequestOptions
+    {
+        return $this->options;
+    }
+
+    /**
+     * Sets the request timeout in seconds.
+     *
+     * @since n.e.x.t
+     *
+     * @param float|null $timeout Timeout in seconds.
+     */
+    public function setTimeout(?float $timeout): void
+    {
+        $options = $this->ensureOptions();
+        $this->options = $options->withTimeout($timeout);
+    }
+
+    /**
+     * Sets the connection timeout in seconds.
+     *
+     * @since n.e.x.t
+     *
+     * @param float|null $timeout Connection timeout in seconds.
+     */
+    public function setConnectTimeout(?float $timeout): void
+    {
+        $options = $this->ensureOptions();
+        $this->options = $options->withConnectTimeout($timeout);
+    }
+
+    /**
+     * Sets whether redirects are automatically followed.
+     *
+     * @since n.e.x.t
+     *
+     * @param bool $allowRedirects Whether redirects should be followed.
+     */
+    public function setAllowRedirects(bool $allowRedirects): void
+    {
+        $options = $this->ensureOptions();
+
+        $this->options = $allowRedirects
+            ? $options->withRedirects()
+            : $options->withoutRedirects();
+    }
+
+    /**
+     * Sets the maximum number of redirects to follow.
+     *
+     * @since n.e.x.t
+     *
+     * @param int|null $maxRedirects Maximum redirects when enabled.
+     */
+    public function setMaxRedirects(?int $maxRedirects): void
+    {
+        $options = $this->ensureOptions();
+        $this->options = $options->withMaxRedirects($maxRedirects);
+    }
+
+    /**
+     * Ensures request options instance exists.
+     *
+     * @since n.e.x.t
+     *
+     * @return RequestOptions The ensured request options instance.
+     */
+    private function ensureOptions(): RequestOptions
+    {
+        if ($this->options === null) {
+            $this->options = new RequestOptions();
+        }
+
+        return $this->options;
+    }
```

Comment:
I think we should get rid of these, for the following reasons:
* The DTOs should be immutable.
* If we wanted setters, they should still be only on `RequestOptions` itself.
* `withOptions` is a sufficient way to achieve this goal.

Setting request options is a lower-level API. So I would say we shouldn't add a bunch of convenience wrappers around it. If we start doing this here, there are other way more first-class APIs where we would need to consider this, but I think it's better to keep the API surface centered to its root.

Reply:
Resolved in 88b0075